haskell/upload-hackage-docs.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io/>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+# Authors: @maksbotan, @gromak
+
+# Use this script if you want to manually upload documentation to Hackage.
+# Sometimes Hackage fails to build your package and you don't want to fix
+# it but want to make your documentation available.
+# You need to specify:
+# • ${PACKAGE_NAME} is a name of a package you are uploading docs for
+# • ${PACKAGE_DIR} is a relative path to that package's directory
+# • ${HACKAGE_USERNAME} and ${HACKAGE_PASSWORD} are your Hackage credentials
+
+set -euo pipefail
+
+stack build --haddock "${PACKAGE_NAME}"
+# Check that package is ok
+stack sdist "${PACKAGE_DIR}"
+# Upload it and ignore errors because it might be already uploaded
+stack upload "${PACKAGE_DIR}" || true
+
+dist=$(stack path --dist-dir 2>/dev/null)
+version=$(stack query locals "${PACKAGE_NAME}" version 2>/dev/null)
+cd "${PACKAGE_DIR}/${dist}/doc/html"
+# Hackage expects documentation in dir with -docs suffix.
+cp -r "${PACKAGE_NAME}" "${PACKAGE_NAME}-${version}-docs"
+# Hackage does not like GNU tar's default format, have to use "ustar".
+tar cvz --format=ustar -f "${PACKAGE_NAME}.tar.gz" "${PACKAGE_NAME}-${version}-docs"
+
+# Alternatively instead of passwords use
+# -H 'Authorization: X-ApiKey TOKEN'
```

Comment:
Maybe add this ability to the script? Check if `HACKAGE_TOKEN` is defined and use it, and only if it's not, resort to passwords.

Reply:
Oh god, I somehow unexpectedly appeared in the situation when I have to write some bash (other than adding comments and quotes) :oh-god-why:

Okay, I will do that (hopefully).

Reply:
Maybe run shellcheck afterwards just in case :p

Reply:
I have `set -u` in the beginning.

Reply:
Okay, done. I have tested it and it even seems to work (both scenarios) :pogchamp:

Reply:
Btw, it would be nice if CI in this repo applied shellcheck to all bash scripts automatically :pled: